mlflow/tensorflow/__init__.py

```diff
@@ -887,6 +968,110 @@ def _log_early_stop_callback_metrics(callback, history, metrics_logger):
+                            )
+
+                        input_example_slice = _extract_n_steps(input_training_data)
+                    elif isinstance(input_training_data, dict):
+                        input_example_slice = {
+                            k: np.take(v, range(0, INPUT_EXAMPLE_SAMPLE_ROWS))
+                            for k, v in input_training_data.items()
+                        }
+                    else:
+                        warnings.warn(
+                            "Tensorflow keras autologging only "
+                            "supports input types of: numpy.ndarray, "
+                            "dict(<key> -> numpy.ndarray), tensorflow.data.Dataset, "
+                            "or tensorflow.keras.utils.Sequence"
+                        )
+
+                    return input_example_slice
+
+                def _infer_model_signature(input_data_slice):
+                    try:
+                        original_stop_training = history.model.stop_training
+                        model_output = history.model.predict(input_data_slice)
+
+                        if (
+                            Version(tensorflow.__version__) <= Version("2.1.4")
+                            and original_stop_training
+                        ):
+                            # For these versions, `stop_training` flag on Model is set to False
+                            # This flag is used by the callback
+                            # (inside ``_log_early_stop_callback_metrics``)
+                            # for logging of early stop metrics. In order for
+                            # that to work, need to force that flag to be True again since doing
+                            # predict on that model sets `stop_training` to false for
+                            # those TF versions
+                            history.model.stop_training = True
+
+                        model_signature = infer_signature(input_data_slice, model_output)
+                    except TypeError as te:
+                        warnings.warn(str(te))
+                        model_signature = None
+                    return model_signature
+
+                input_example, signature = resolve_input_example_and_signature(
+                    _get_input_data_slice,
+                    _infer_model_signature,
+                    log_input_examples,
+                    log_model_signatures,
+                    _logger,
+                )
+
+                mlflow.keras.log_model(
+                    keras_model=history.model,
+                    artifact_path="model",
+                    input_example=input_example,
+                    signature=signature,
+                    registered_model_name=get_autologging_config(
+                        FLAVOR_NAME, "registered_model_name", None
+                    ),
+                )
+        except Exception as e:
+            warnings.warn(f"Failed to autolog keras model: {str(e)}")
```

Comment:
We shouldn't try / catch the model logging procedure. If overall model logging fails, the run should be marked as "failed". The autologging safe patch framework handles this automatically. We can just remove the outer try/catch here.

Reply:
done. thx